homeassistant/components/ruckus_unleashed/__init__.py

```diff
@@ -24,35 +26,45 @@
 )
 from .coordinator import RuckusUnleashedDataUpdateCoordinator
 
+_LOGGER = logging.getLogger(__package__)
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Ruckus Unleashed from a config entry."""
+
     try:
-        ruckus = await Ruckus.create(
+        ruckus = AjaxSession.async_create(
             entry.data[CONF_HOST],
             entry.data[CONF_USERNAME],
             entry.data[CONF_PASSWORD],
         )
-    except ConnectionError as error:
-        raise ConfigEntryNotReady from error
+        await ruckus.login()
+    except (ConnectionRefusedError, ConnectionError) as conerr:
+        raise ConfigEntryNotReady from conerr
+    except AuthenticationError as autherr:
+        raise ConfigEntryAuthFailed from autherr
 
     coordinator = RuckusUnleashedDataUpdateCoordinator(hass, ruckus=ruckus)
 
     await coordinator.async_config_entry_first_refresh()
 
-    system_info = await ruckus.system_info()
+    system_info = await ruckus.api.get_system_info()
 
     registry = dr.async_get(hass)
-    ap_info = await ruckus.ap_info()
-    for device in ap_info[API_AP][API_ID].values():
+    aps = await ruckus.api.get_aps()
+    for access_point in aps:
+        _LOGGER.debug("AP [%s] %s", access_point[API_AP_SERIALNUMBER], entry.entry_id)
         registry.async_get_or_create(
             config_entry_id=entry.entry_id,
-            connections={(dr.CONNECTION_NETWORK_MAC, device[API_MAC])},
-            identifiers={(dr.CONNECTION_NETWORK_MAC, device[API_MAC])},
+            connections={(dr.CONNECTION_NETWORK_MAC, access_point[API_AP_MAC])},
+            identifiers={(DOMAIN, access_point[API_AP_SERIALNUMBER])},
```

Comment:
Could you please tell me why you changed the identifier?

Reply:
My guess was that the unique AP serial number was a better identifier, given I might want to use the MAC for something else (e.g. to allow disabling/enabling ethernet ports).

Happy to revert this change if I got this wrong.

Reply:
Changing the identifier also requires adding a migration for it. Otherwise, we would end up with orphan devices.
This PR is to change the dependency, and we should keep it as small as possible.
Changing the identifier is a feature and should go, if desired, into a new PR.

Please revert this change. After this PR gets merged, you can create a new one, where you change the identifier and add migration for it.

Reply:
reverted